cpp/inc/bond/stream/input_buffer.h

```diff
@@ -126,38 +128,42 @@ class InputBuffer
         {
             EofException(sizeof(uint8_t));
         }
-        
+
         value = static_cast<const uint8_t>(_blob.content()[_pointer++]);
     }
 
-    
+
     template <typename T>
     void Read(T& value)
     {
+        BOOST_STATIC_ASSERT(bond::is_arithmetic<T>::value || bond::is_enum<T>::value);
+
         if (sizeof(T) > _blob.length() - _pointer)
         {
             EofException(sizeof(T));
         }
-        
-        value = *reinterpret_cast<const T*>(_blob.content() + _pointer);
+
+        const void* const src = _blob.content() + _pointer;
+        std::memcpy(&value, src, sizeof(T));
```

Comment:
TODO: see whether on x86 there's any performance regression due to this change in reading. If there is, guard the unaligned access like is done in output_buffer.